controllers/containerimagebuild_controller.go

```diff
@@ -0,0 +1,129 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	forgev1alpha1 "github.com/dominodatalab/forge/api/v1alpha1"
+	"github.com/dominodatalab/forge/pkg/container"
+	"github.com/dominodatalab/forge/pkg/container/config"
+)
+
+// ContainerImageBuildReconciler reconciles a ContainerImageBuild object
+type ContainerImageBuildReconciler struct {
+	client.Client
+	Log      logr.Logger
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+	Builder  container.RuntimeBuilder
+}
+
+// +kubebuilder:rbac:groups=forge.dominodatalab.com,resources=containerimagebuilds,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=forge.dominodatalab.com,resources=containerimagebuilds/status,verbs=get;update;patch
+
+func (r *ContainerImageBuildReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("containerimagebuild", req.Name)
+
+	var result ctrl.Result
+	var build forgev1alpha1.ContainerImageBuild
+
+	// attempt to load resource by name and ignore not-found errors
+	if err := r.Get(ctx, req.NamespacedName, &build); err != nil {
+		log.Error(err, "Unable to find resource")
+		return result, client.IgnoreNotFound(err)
+	}
+
+	// ignore resources that have been processed on start
+	if len(build.Status.State) != 0 {
+		log.Info("Skipping resource", "state", build.Status.State)
+		return result, nil
+	}
+
+	// mark resource status and update before launching build
+	build.Status.State = forgev1alpha1.Building
+	build.Status.BuildStartedAt = &metav1.Time{Time: time.Now()}
+
+	if err := r.updateResourceStatus(ctx, log, &build); err != nil {
+		return result, err
+	}
+
+	// construct build directives and dispatch operation
+	opts := config.BuildOptions{
+		ImageName:        build.Spec.ImageName,
+		Context:          build.Spec.Context,
+		RegistryURL:      build.Spec.PushRegistry,
+		NoCache:          build.Spec.NoCache,
+		Labels:           build.Spec.Labels,
+		BuildArgs:        build.Spec.BuildArgs,
+		CpuQuota:         build.Spec.CpuQuota,
+		Memory:           build.Spec.Memory,
+		InsecureRegistry: true,
```

Comment:
Etienne mentioned this. I think we _should_ support them but the question is whether we add that now or after we integration. Thoughts?

Reply:
The only reason I say that is because we have a custom registry cert for the internal registry in production installs ([domino_cert_mgr](https://github.com/cerebrotech/platform-apps/tree/develop/src/domino_cert_mgr) is the one that ferries the cert around on the host at the moment).

Reply:
I see. Then it sounds like this is a _required_ feature prior to going to production. Let's discuss this later today.